### PR TITLE
Add maintenance mode styles and markup for widgets

### PR DESCRIPTION
**Pull request by thevoiceofzeke**
**In this PR**:
- Added CSS styles for maintenance mode "overlay" (meaning: A thing that goes on top of widgets)
- Added HTML markup for maintenance mode
- Added maintenanceMode attribute to widget creator templates
- Added info about maintenance mode to the widget documentation 
- Added working widget docs link to widget creator page
- Add `.editorconfig` to ensure consistent code style (same as uw-frame)

*Notes: I'm confident it'll work, having tested it with the widget creator, but it should still be tested with a real widgetConfig for a real widget. Also, please ignore the whitespace gore in widgetController. I was desperate to add some missing semi-colons and this project has no code style configuration :/*

### Screenshots
screen shot 2017-01-26 at 12 47 06 pm
screen shot 2017-01-26 at 12 51 58 pm

### CODE REVIEW

File: angularjs-portal-home/src/main/webapp/css/widget.less [hunk 1]
```diff
@@ -281,4 +281,33 @@ weather {
 .widget-middle {
   width : 315px;
   margin : auto;
+}
+
+/* MAINTENANCE MODE */
+.widget-frame {
+  .overlay__maintenance-mode {
+	position: absolute;
```
Comment (apetro) [inline angularjs-portal-home/src/main/webapp/css/widget.less]: Prefer spaces over tabs.

**Comment by thevoiceofzeke** [comment]
@vertein If we're going to have a de facto code style, we need a `.editorconfig` file with the rules. I suggest copying the one from the uw-frame project. I work on too many projects with too many different preferences to change my settings every time I change projects :/

File: angularjs-portal-home/src/main/webapp/my-app/layout/widget/controllers.js [hunk 2]
```diff
@@ -399,7 +400,7 @@ define(['angular'], function(angular){
             widgetType : 'rss',
             title: 'RSS Widget',
             jsonSample: false,
-            widgetConfig : {lim : 6, target : '' ,showdate: true, titleLim: 40 , dateFormat: 'MM-dd-yyyy', showShowing: true},
+            widgetConfig : {lim : 6, target : '' ,showdate: true, titleLim: 40 , dateFormat: 'MM-dd-yyyy', showShowing: true, maintenanceMode: false},
```
Comment (apetro) [inline angularjs-portal-home/src/main/webapp/my-app/layout/widget/controllers.js]: I'm missing something. Maintenance mode flagging is going to come through as `widgetConfig`? How does this relate to flopping apps over into maintenance lifecycle state server-side?

File: angularjs-portal-home/src/main/webapp/my-app/layout/widget/partials/widget-card.html [hunk 3]
```diff
@@ -1,9 +1,19 @@
 <md-card class="widget-frame" id="portlet-id-{{portlet.nodeId}}" aria-label="{{ portlet.title }} widget">
 
+	<!-- MAINTENANCE MODE OVERLAY -->
+	<div class="overlay__maintenance-mode" ng-if="portlet.widgetConfig.maintenanceMode">
+		<div class="maintenance-content">
+			<p>
+				<md-icon class="md-warn">warning</md-icon>
+			</p>
+			<p>This app is undergoing scheduled maintenance and is temporarily unavailable.</p>
```
Comment (apetro) [inline angularjs-portal-home/src/main/webapp/my-app/layout/widget/partials/widget-card.html]: I'd drop the "scheduled" so we're more free to use this maintenance mode in a wider variety of situations.

File: angularjs-portal-home/src/main/webapp/my-app/layout/widget/partials/widget-card.html [hunk 4]
```diff
@@ -12,7 +22,7 @@
     <md-button class="widget-action widget-remove md-icon-button"
                aria-label="remove {{ portlet.title }} widget from your home screen"
                ng-click="widgetCtrl.removePortlet(portlet.nodeId, portlet.title)"
-               ng-hide="GuestMode || cantRemove">
+               ng-hide="GuestMode || cantRemove || portlet.widgetConfig.maintenanceMode">
```
Comment (apetro) [inline angularjs-portal-home/src/main/webapp/my-app/layout/widget/partials/widget-card.html]: Nice!

File: docs/widgets.md [hunk 5]
```diff
@@ -238,17 +238,25 @@ By doing just this we were able to generate:
 
 ## Other Configuration
 
+#### Launch button text
```
Comment (apetro) [inline docs/widgets.md]: Suggestion: use headings semantically. The first heading under `##` should be `###`.

**Pull request merged**